hermes/file_gen_test.go

```diff
@@ -47,13 +28,32 @@ func TestChecksum(t *testing.T) {
 	}
 }
 
-// Test function for the generateFileName function in file_gen.go
+type FileNameTestUnit struct {
+	id       int
+	checksum string
+	want     string
+}
+
+var test_unit_table [5]FileNameTestUnit
+
 func TestFileName(t *testing.T) {
-	completeTestUnitTable()
+	test_unit_table[0].id = 1
+	test_unit_table[0].checksum = "abba"
+	test_unit_table[0].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[0].id, test_unit_table[0].checksum)
+	test_unit_table[1].id = 11
+	test_unit_table[1].checksum = "abba"
+	test_unit_table[1].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[1].id, test_unit_table[1].checksum)
+	test_unit_table[2].id = 21
+	test_unit_table[2].checksum = "abba"
+	test_unit_table[2].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[2].id, test_unit_table[2].checksum)
+	test_unit_table[3].id = 31
+	test_unit_table[3].checksum = "abba"
+	test_unit_table[3].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[3].id, test_unit_table[3].checksum)
+	test_unit_table[4].id = 41
+	test_unit_table[4].checksum = "abba"
+	test_unit_table[4].want = fmt.Sprintf("Hermes_%02d_%v", test_unit_table[4].id, test_unit_table[4].checksum)
 	for i := 0; i < 5; i++ {
-		file := HermesFile{}
-		file.generateFileName(test_unit_table[i].id, test_unit_table[i].checksum)
-		got := file.name
+		got := generateFileName(test_unit_table[i].id, test_unit_table[i].checksum)
```

Comment:
Test the public API of the package instead of individual functions.
The test should exercise the NewHermesFile() function and verify that the resulting HermesFile instances meet the expectations. If they do, then implicitly these lower level functions must work correctly.
The problem with testing these lower-level functions:
- it ties the tests to implementation details, making it harder to change the implementation (it forces you to change the tests as well)
- it doesn't exercise all the layers of the package, leaving some code untested